arelle/ModelInstanceObject.py

```diff
@@ -513,7 +513,7 @@ def isDuplicateOf(self, other, topLevel=True, deemP0Equal=False, unmatchedFactsS
             if self.modelXbrl == other.modelXbrl and self.parentElement != other.parentElement:
                 return False
             if not (self.context.isEqualTo(other.context,dimensionalAspectModel=False) and
-                    (not self.isNumeric or self.unit.isEqualTo(other.unit))):
+                    (not self.isNumeric or (self.unit is not None and self.unit.isEqualTo(other.unit)))):
```

Comment:
I wonder whether a numeric fact should be considered duplicate if both of them are missing a unit.

Reply:
Austin was first :-).